include/surface_perception/surface_finder.h

```diff
@@ -0,0 +1,133 @@
+  void set_cloud(const pcl::PointCloud<pcl::PointXYZRGB>::Ptr& cloud);
+
+  /// \brief Set the indices of the input point cloud.
+  ///
+  /// \param[in] cloud_indices The indices of the input point cloud.
+  void set_cloud_indices(const pcl::PointIndices::Ptr cloud_indices);
+
+  /// \brief Set the angle of the surface compare to horizontal surface
+  ///
+  /// Because the current state algorithm only search for horizontal surfaces,
+  /// the angle set through this function is not used in surface detection.
+  ///
+  /// \param[in] angle_tolerance The maximum angle difference between a
+  ///  surface candidate against a horizontal surface.
+  void set_angle_tolerance(const double& angle_tolerance);
+
+  /// \brief Set the maximum distance for a point to be considered part of
+  /// surfaces.
+  ///
+  /// \param[in] max_point_distance The maximum distance between a point and
+  ///  a plane that represents the surface.
+  void set_max_point_distance(const double& max_point_distance);
+
+  /// \brief Set the minimum number of iterations for the algorithm to find
+  /// surfaces
+  ///
+  /// The algorithm is designed to run at least the given number of iteration or
+  /// finding the specified minimum number of surfaces in a given scene.
+  ///
+  /// \param[in] min_iteration The minimum number of iteration required before
+  ///  the algorithm can stop search for surfaces.
+  void set_min_iteration(const size_t& min_iteration);
+
+  /// \brief Set the minimum amount of points contained by a surface candidate.
+  ///
+  /// The amount of points for a surface is indicated by the number of points
+  /// reside within the maximum distance of a plane. As the algorithm explores
+  /// surfaces, the surface that have points less that required amount will be
+  /// considered a candidate.
+  ///
+  /// \param[in] surface_point_threshold The minimum number of points a surface
+  ///  candidate must have.
+  void set_surface_point_threshold(const size_t& surface_point_threshold);
+
+  /// \brief Find the horizontal surfaces in a point cloud scene
+  ///
+  /// The algorithm attempts to surfaces in a point cloud scene and terminate if
+  /// all of the following conditions are met:
+  ///  1. The algorithm finds the minimum number of surface required.
+  ///  2. The algorithm finishes the specified number of iteration.
+  ///
+  /// Once the surfaces are found, the surface i has:
+  ///  1. indices as indices_iternals[i]
+  ///  2. coefficients as coeffs[i]
+  ///  3. point cloud history as history[i]
+  ///
+  /// \param[in] min_surface_amount The minimum number of surface must be found
+  /// before
+  ///  the algorithm stops.
+  /// \param[in] max_surface_amount The maximum number of surface in the output.
+  /// \param[out] indices_internals The indices for of each output surface.
+  /// \param[out] coeffs The coefficients of planes that represent each surface.
+  /// \param[out] history The concatenated point cloud that represents the
```

Comment:
You should make a note that the history is optional here, or remove it entirely.

I actually don't think the history is necessary. It was useful for debugging and development, but probably shouldn't be in the public API.

I would still keep the SurfaceHistoryRecorder class around for future debugging, but I would leave it out of the public API.

Reply:
Completed